src/components/PieChart/PieChart.tsx

```diff
@@ -0,0 +1,195 @@
+    };
+
+    private _radius: any;
+    private _focus: any;
+    private _arc: any;
+    private _textCoordinates: number[];
+
+    constructor() {
+        super();
+    }
+
+    componentDidMount() {
+        this.draw();
+    }
+
+    componentDidUpdate() {
+        this.redraw();
+    }
+
+    private redraw() {
+        d3.select('.svg-container').remove();
+        this.draw();
+    }
+
+    public render() {
+        return (<div className={'pie-chart-component'}>
+                    <Label className={'title'}>{this.props.title}</Label>
+                    <Label className={'text'}>{this.props.text}</Label>
+                    <div className={'pie-chart-container'} ref={'container'}></div>
+                </div>);
+    }
+
+    private draw() {
+        this._radius = (this.props.width - 2) / 4;
+        const svg = this.createContainer();
+        const pie = this.createPie();
+        const arc = this.createArc();
+
+        let g = svg.selectAll('.arc')
+            .data(pie(this.props.data))
+            .enter()
+            .append('g')
+            .attr('class', 'arc');
+
+        g.append('path').attr('d', (arc as any))
+            .attr('class', (d) => this.calculateClass(d.data));
+
+        g = svg.selectAll('.arc-text')
+            .data(pie(this.props.data))
+            .enter()
+            .append('g')
+            .attr('class', 'arc-text');
+
+        g.append('text')
+            .attr('transform',
+            (d) => { this._textCoordinates = arc.centroid(d); return 'translate(' + arc.centroid(d) + ')' })
+            .text((d) => d.data.value + '%')
+            .style('font-size', (this._radius / 3))
+            .attr('text-anchor', 'middle')
+            .attr('class', 'percentage-label')
+            .on('mouseover', (d) => this.showTooltip(d))
+            .on('mouseout', () => this._focus.style('display', 'none'));
+
+            this.createTooltip(svg);
+    }
+
+    private showTooltip(d: any) {
+        const coordinates = this._arc.centroid(d);
+
+        this._focus.style('display', 'block');
+
+        this._focus.select('.tip-rect')
+            .attr('transform',
+            'translate(' + (coordinates[0] - this._radius * 1.5) + ',' + (coordinates[1] - this._radius * (3 / 2)) + ')');
+
+        this._focus.select('.tip-pol')
+            .attr('transform',
+            'translate(' + (coordinates[0] - this._radius) + ',' + (coordinates[1] - this._radius * (3 / 2)) + ')');
+
+        this._focus.select('text.tooltip-text')
+            .attr('transform',
+            'translate(' + (coordinates[0] - this._radius * 1.5) + ',' + (coordinates[1] - this._radius * (5 / 4)) + ')')
+            .text(d.data.text);
+    }
+
+    private calculateClass(d: DataType) {
+        if (d.label === 'free') { return 'disk-usage-free'; }
+        
+        let className;
+
+        if (this.props.statusInterval) {
+            const okInterval = this.props.statusInterval.filter((d) => d.status === 'ok')[0];
```

Comment:
We could make component drawing some random (or specified) colors that aren't tied up to classes or anything, so every part of chart will be in some of specified color.